src/ImageSharp.Web/Middleware/ImageSharpMiddlewareOptions.cs

```diff
@@ -24,28 +24,17 @@ public class ImageSharpMiddlewareOptions
                 return Task.CompletedTask;
             }
 
-            // It's a good idea to have this to provide very basic security.
```

Comment:
I'm gonna scrap all of this and introduce optional HMAC functionality instead.

Reply:
Although I would love to see built-in HMAC validation support, that would be a major feature and something that would only work if it's fully integrated in the consuming project (all image processing URLs would need to be generated with logic that adds the HMAC hash/token).

This is also something that Orchard Core has [already implemented themselves](https://github.com/OrchardCMS/OrchardCore/blob/c8b2954e038aaa54963b60bc6cfe193cfea54799/src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaImageSharpConfiguration.cs#L36-L63) (maybe check with @deanmarcussen).

Having some default validation on the maximum width/height actually makes sense, as values >4000 are still valid, but would easily drain your CPU, memory and cache/disk resources.